junifer/preprocess/base.py

```diff
@@ -0,0 +1,174 @@
+"""Provide abstract base class for preprocessor."""
+
+# Authors: Federico Raimondo <f.raimondo@fz-juelich.de>
+#          Synchon Mandal <s.mandal@fz-juelich.de>
+# License: AGPL
+
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Union, Tuple, Any
+
+
+from ..pipeline import PipelineStepMixin
+from ..utils import logger, raise_error
+
+
+class BasePreprocessor(ABC, PipelineStepMixin):
+    """Provide abstract base class for all preprocessors.
+
+    Parameters
+    ----------
+    on : str or list of str, optional
+        The kind of data to apply the preprocessor to. If None,
+        will work on all available data (default None).
+    """
+
+    def __init__(
+        self,
+        on: Optional[Union[List[str], str]] = None,
```

Comment:
Having a `name` argument here would allow us to save it to the `meta`. What do you think?

Reply:
Name is only for markers, as there can be many markers on the "pipeline". The "name" in this case is "preprocess"

Reply:
Fair enough.